processor/src/main/java/org/mapstruct/ap/util/MapperConfig.java

```diff
@@ -73,11 +74,25 @@ private MapperConfig(MapperPrism mapperPrism) {
     }
 
     public List<TypeMirror> uses() {
-        Set<TypeMirror> uses = new HashSet<TypeMirror>( mapperPrism.uses() );
```

Comment:
Wouldn't using a `LinkedHashSet` do the trick, avoiding the manual "add-all-once" handling while still guaranteeing a stable iteration order?